File: src/main/java/nl/haltedata/check/model/NetexRouteQuay.java
package nl.haltedata.check.model;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class NetexRouteQuay {
    String routeId;
    Long quayIndex;
    String lineNumber;
    String quayCode;
    String quayName;
    String stopSideCode;
    String stopplaceCode;
    String quayLocationType;
    
    @Override
    public String toString() {
        var sb = new StringBuilder();
        sb.append(quayCode).append(" - ").append(quayName);
        if (stopSideCode != null) {
            sb.append(" (").append(stopSideCode).append(")");
        }
        return sb.toString();
    }
}

File: frontend/src/app/quays/netex-route-quay.ts
export default class NetexRouteQuay {
    public pointOnRouteId!: string;
    public routeId!: string;
    public quayIndex!: number;
    public quayCode!: string;
    public town!: string;
    public quayName!: string;
    public stopSideCode!: string;
    public stopPlaceCode!: string;

    public get name(): string {
        return this.quayName + (this.stopSideCode ? " (" + this.stopSideCode + ")" : "");
    }
    
    public get uov(): string {
        var uov = this.town.toLowerCase() + "-" + this.quayName.toLowerCase();
        return uov.replaceAll(" ", "-").replaceAll("'", "").replaceAll("+", "");
    }

    public get _9292(): string {
        var town = this.town.toLowerCase().replaceAll(" ", "-").replaceAll("'", "");
        var quay = this.quayName.toLowerCase().replaceAll(" ", "-").replaceAll("'", "").replaceAll("+", "").replaceAll("/", "-");
        return town + "/bushalte-" + quay;
    }

}
